dcos_launch/terraform.py

```diff
@@ -0,0 +1,100 @@
+from dcos_launch import util
+import yaml
+import os
+import json
+import shutil
+import uuid
+
+# TODO pinning terraform-dcos is not enough because its tf_dcos_core dependency is not pinned. Create a branch on
+# terraform-dcos that pins tf_dcos_core and replace TERRAFORM_DCOS_VERSION here by that branch name. Regularly update
+# that branch with master.
+TERRAFORM_DCOS_VERSION = 'master'
+TERRAFORM_DCOS_EE_VERSION = 'service_credentials_gcp2'
+
+
+class TerraformLauncher(util.AbstractLauncher):
+    def __init__(self, config: dict, env=None):
+        self.config = config
+        self.init_dir = config.get('init_dir')
+        if not self.init_dir:
+            self.init_dir = 'terraform-init-' + uuid.uuid4().__str__()
+            self.config['init_dir'] = self.init_dir
+        self.init_dir = os.path.abspath(self.init_dir)
```

Comment:
if the user supplies this, they may decide to use a relevant path, perhaps use this to avoid that problem: https://github.com/dcos/dcos-launch/blob/master/dcos_launch/config.py#L14

Reply:
I actually would never want a user to supply this. It's only meant to be generated the 1st time, written to cluster_info.json, then re-used from there. I'll throw in a regex as a requirement and that should fix it.